core/src/main/java/io/questdb/metrics/MemCappedQueryTraceQueue.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.metrics;
+
+import io.questdb.mp.ConcurrentQueue;
+import io.questdb.std.ObjectFactory;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class MemCappedQueryTraceQueue {
```

Comment:
this looks like a double-whammy. A bit of a background:

We already had fixed size queues, RingQueue, this provides natural pushback from backpressure. This queue became inconvenient in situations where messages must not be discarded. Such as transactions.

To deal with this shortcoming Alex introduced an unbounded queue, which is simpler to use. E.g. no workaround needed if queue is full.

This class takes unbounded queue and converts it into a bounded queue, which is already exists elsewhere (RingQueue).

On top of that, we should not be bounding this pipeline prematurely. Until it becomes a problem.

Reply:
The bound is not on element count, but on total memory used by the strings. We got concerned with the possibility of lots of large query strings being retained on the queue.

Reply:
I would drop this complexity until the time we see it become a problem. Unpicking this query is very fast, writer can smash in 10M rows/s, which way higher performance than that of a SQL parser or network protocols.

Reply:
Resolved in 98b818591e2bb7859a146261b3dd75a67ed84dba